packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -858,8 +723,14 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         if (!tree) {
             tree = await getWithRetryForTokenRefresh(async (options) => {
                 const storageToken = await this.getStorageToken(options, "ReadCommit");
-
-                const response = await fetchSnapshot(this.snapshotUrl!, storageToken, id, this.fetchFullSnapshot, this.logger, this.epochTracker);
+                const snapshotDownloader = async (url: string, fetchOptions: {[index: string]: any}) => {
+                    return this.epochTracker.fetchAndParseAsJSON<IOdspSnapshot>(
+                        url,
+                        fetchOptions,
+                        "snapshotTree",
+                    );
+                };
+                const response = await fetchSnapshot(this.snapshotUrl!, storageToken, id, this.fetchFullSnapshot, this.logger, snapshotDownloader);
```

Comment:
This is probably beyond this change, but should this API exists at all? I.e. should we call here newly created fetchLatestSnapshotCore() instead (and figure out how to support latest vs. specific version)? I believe this is used only by summarizer today when it switched to full summary and needs to refresh its knowledge about prior snapshot, so perf (POST vs. GET) is probably not that critical.
If you feel like it's possible and worth following up, please open an issue

Reply:
Will take a look and open a issue for this.